### PR TITLE
Add replit to sandboxes [fixes: #4820]

### PR DESCRIPTION
**Pull request by notjustinshaw**
## Description

Added Replit to the list of sandboxes in the website. Removed the Remix promo banner (since Remix is no longer the only sandbox listed that is also an end-to-end development environment).

## Related Issue
#4820

### CODE REVIEW

File: src/pages/developers/learning-tools.js [hunk 1]
```diff
@@ -80,6 +80,15 @@ const LearningToolsPage = ({ data }) => {
...
   const sandboxes = [
+    {
+      name: "Replit",
+      description: "page-learning-tools-replit-description",
+      url: "https://replit.com/@replit/Solidity-starter-beta",
+      image: getImage(data.replit),
+      alt: "page-learning-tools-replit-logo-alt",
+      background: "#11131D",
+      subjects: ["Solidity", "web3"],
+    },
```
Comment (notjustinshaw) [inline src/pages/developers/learning-tools.js:91]: @sorenrood can you double check that this is correct?

**Review by notjustinshaw — COMMENTED** [review]
I got most of the heavy stuff moved over, but there are still a few things left to do. First, we'll have Soren at Replit come in and replace the placeholder copy I wrote with something better. Once that's done, we'll need someone from the web team to come and help guide this PR through the translation steps. Then we should be ready to land this PR on the dev branch.

There also might need to be a side conversation about the Remix banner that I removed. Maybe people still want that. I figured that since Replit also offers an end-to-end development suite it would be impartial to support one IDE over another. Though I could easily see supporters of Remix arguing that their IDE is a first-party service provided by the eth team, so maybe it should get its own banner. I don't know.

File: src/pages/developers/learning-tools.js (hunk 1 above)
Comment (sorenrood) [reply src/pages/developers/learning-tools.js:91]: Link looks good. Can we use `#0f1524`?

File: src/intl/en/page-developers-learning-tools.json [hunk 2]
```diff
@@ -29,8 +29,9 @@
   "page-learning-tools-questbook-description": "Self paced tutorials to learn Web 3.0 by building",
   "page-learning-tools-questbook-logo-alt": "Questbook Logo",
   "page-learning-tools-remix-description": "Develop, deploy and administer smart contracts for Ethereum. Follow tutorials with the Learneth plugin.",
-  "page-learning-tools-remix-description-2": "Remix is not just a sandbox. Many developers write, compile and deploy their smart contracts using Remix.",
   "page-learning-tools-remix-logo-alt": "Remix logo",
+  "page-learning-tools-replit-description": "An integrated development environment for Ethereum with hot reload, error checking, and first-class testnet support.",
+  "page-learning-tools-replit-logo-alt": "Replit logo",
```
[the remark this answers is not in the document]
Comment (sorenrood) [reply src/intl/en/page-developers-learning-tools.json:34]: Copy looks good to me! Thanks Justin.

[the remark this answers is not in the document]
Comment (sorenrood) [reply src/intl/en/page-developers-learning-tools.json:34]: Can we actually replace the word "integrated" with "customizable"? I want it to be clear that the template is hackable.

File: src/pages/developers/learning-tools.js [hunk 3]
```diff
@@ -355,6 +361,9 @@ export const query = graphql`
     remix: file(relativePath: { eq: "dev-tools/remix.png" }) {
       ...learningToolImage
     }
+    replit: file(relativePath: { eq: "dev-tools/replit.png" }) {
+      ...learningToolImage
+    }
```
[the remark this answers is not in the document]
Comment (sorenrood) [reply src/pages/developers/learning-tools.js:366]: Yup, logo is good here! 👍🏻

**Comment by sorenrood** [comment]
Thanks for all of your help with this, @notjustinshaw!

**Review by sorenrood — APPROVED** [review]

File: src/intl/en/page-developers-learning-tools.json (hunk 2 above)
[the remark this answers is not in the document]
Comment (sorenrood) [reply src/intl/en/page-developers-learning-tools.json:34]: "A customizable development environment for Ethereum with hot reloading, error checking, and first-class testnet support."

**Review by sorenrood — APPROVED** [review]

File: src/pages/developers/learning-tools.js (hunk 1 above)
Comment (notjustinshaw) [reply src/pages/developers/learning-tools.js]: done.

File: src/intl/en/page-developers-learning-tools.json (hunk 2 above)
[the remark this answers is not in the document]
Comment (notjustinshaw) [reply src/intl/en/page-developers-learning-tools.json]: done

File: src/pages/developers/learning-tools.js (hunk 3 above)
[the remark this answers is not in the document]
Comment (notjustinshaw) [reply src/pages/developers/learning-tools.js:366]: awesome!

**Review by sorenrood — APPROVED** [review]

**Comment by notjustinshaw** [comment]
And then if we can we should also add @sorenrood for docs, content

cc @all-contributors

**Comment by corwintines** [comment]
Id like to randomize this list, but overall looking good. Ill push a commit up for this.

**Comment by corwintines** [comment]
@notjustinshaw @sorenrood im just testing this out, and when I go to open the link you provided for the url (https://replit.com/@replit/Solidity-starter-beta) and try to run the example I am running into issues. I have attached an image below:

Screen Shot 2022-01-03 at 2 37 39 PM

**Comment by notjustinshaw** [comment]
Hmm. Looks like the "run example" button doesn’t work for logged-out users, that’s probably not the expected behavior. I tried this on a private browser and got the same thing, but if you’re logged in you get a nice page and it all works as expected.

Even if you’re logged-out, you can click "open website" and the UI works, but you have to create an account for the coding environment to show up correctly I guess.
On Jan 3, 2022, 1:39 PM -0800, Corwin Smith ***@***.***>, wrote:
	@notjustinshaw @sorenrood im just testing this out, and when I go to open the link you provided for the url ***@***.***/Solidity-starter-beta) and try to run the example I am running into issues. I have attached an image below:

**Comment by sorenrood** [comment]
> that’s probably not the expected behavior

Correct, looks like this is a recent regression. We're working on a fix now. Thanks for pointing this out. Will update shortly.

The _expected_ behavior is that anonymous users can run the template. If a user wants to fork the template, they'll be prompted to make an account.

**Comment by sorenrood** [comment]
@corwintines @notjustinshaw the issue should be fixed! Just tried in private browser and it's working as-intended.

**Comment by minimalsm** [comment]
Hey @notjustinshaw @sorenrood thanks for working on this.

On removing the Remix callout: we want to make it clear to users that code sandboxes (Remix and Replit) can both be used as an end-to-end development tool. With that as our goal I think we should:

1. [Reverse this commit](https://github.com/ethereum/ethereum-org-website/pull/4921/commits/1b21fc5fffab490f2b18cbd9379e45f37274ce57) to re-add our original translations 
2. Change the callout/promo banner to include Replit, something like: "Remix and Replit aren't just sandboxes—developers can write, compile and deploy their smart contracts using them."
3. Once this is done we'll upload it to our translation pipeline.

Doing it this way preserves current translations for non-English speakers until we get the new copy translated.

File: src/intl/en/page-developers-learning-tools.json [hunk 4]
```diff
@@ -29,7 +29,7 @@
   "page-learning-tools-questbook-description": "Self paced tutorials to learn Web 3.0 by building",
   "page-learning-tools-questbook-logo-alt": "Questbook Logo",
   "page-learning-tools-remix-description": "Develop, deploy and administer smart contracts for Ethereum. Follow tutorials with the Learneth plugin.",
-  "page-learning-tools-remix-description-2": "Remix is not just a sandbox. Many developers write, compile and deploy their smart contracts using Remix.",
+  "page-learning-tools-end-to-end-callout-description": "Remix and Replit aren't just sandboxes—developers can write, compile and deploy their smart contracts using them.",
```
Comment (notjustinshaw) [inline src/intl/en/page-developers-learning-tools.json:32]: I updated the english description here as requested. I also updated the tag name from `page-learning-tools-remix-description-2` to `page-learning-tools-end-to-end-callout-description` to reflect the fact that this callout is about end-to-end development environments (both replit and remix).

**Comment by notjustinshaw** [comment]
We're ready for you to take another look 🥳  @minimalsm @sorenrood

**Comment by minimalsm** [comment]
Hey @notjustinshaw thanks for making these changes 😬 

We're getting a fresh deploy of the site out right now so in the interest of getting this live before the weekend I quickly pulled the changes into #5016 and it should be live [on this page](https://ethereum.org/en/developers/learning-tools/) within the next ~30 mins 🎉

Not context you would have had but changing the key `page-learning-tools-remix-description-2` in the JSON file in this way would cause a synchronisation problem with our files on GitHub that is a bit of a headache for us 😅 I've updated the copy but kept the old key for now until we get an opportunity to do a sweep of these.

Thanks again for working on this. Excited to see Replit being used to develop dapps ☺️❤️

**Comment by minimalsm** [comment]
@all-contributors please add @notjustinshaw @sorenrood for code, docs, ideas

**Comment by minimalsm** [comment]
Oh and we'll be announcing our 2022 contributor POAP soon, you can both [claim your POAP](https://ethereum.org/en/contributing/#poap) pre-emptively by following these instructions 🥳!

**Comment by minimalsm** [comment]
@all-contributors please add @sorenrood for code, docs, ideas

**Comment by sorenrood** [comment]
Hmm. Any reason Replit isn't on the page yet?

**Pull request closed without merging**